### PR TITLE
update selfplay to write 5% of games to a holdout directory

### PR DESCRIPTION
**Pull request by amj**
- Changes the signature on preprocessing.make_dataset_from_selfplay to return the tf_examples instead of write a file
- sets a default holdout directory and makes the appropriate changes in rl_loop.py
- currently not an argument, nor is the amount of holdout a parameter.  Should it be?

### CODE REVIEW

File: preprocessing.py [hunk 1]
```diff
@@ -161,15 +161,15 @@ def get_input_tensors(batch_size, tf_records, num_repeats=None,
 # End-to-end utility functions
 
 
-def make_dataset_from_selfplay(data_extracts, tf_record):
+def make_dataset_from_selfplay(data_extracts):
     '''
+    Returns a set of tf.Examples.
```
Comment (brilee) [inline preprocessing.py]: return an iterable of tf.Example

File: rl_loop.py [hunk 2]
```diff
@@ -28,6 +28,7 @@
 BASE_DIR = "gs://{}".format(BUCKET_NAME)
 MODELS_DIR = os.path.join(BASE_DIR, 'models')
 SELFPLAY_DIR = os.path.join(BASE_DIR, 'games')
```
Comment (brilee) [inline rl_loop.py]: While we're rejiggering our folder naming scheme, can we rename the selfplay dir? "games" is pretty nondescriptive.

File: main.py [hunk 3]
```diff
@@ -138,8 +139,15 @@ def selfplay(
     with gfile.GFile(os.path.join(output_sgf, '{}.sgf'.format(output_name)), 'w') as f:
         f.write(player.to_sgf())
 
-    fname = os.path.join(output_dir, "{}.tfrecord.zz".format(output_name))
-    preprocessing.make_dataset_from_selfplay(game_data, fname)
+    tf_examples = preprocessing.make_dataset_from_selfplay(game_data)
+
+    # Hold out 5% of games for evaluation.
```
Comment (brilee) [inline main.py]: make 0.05 a flag?

File: rl_loop.py (hunk 2 above)
Comment (amj) [reply rl_loop.py]: Done.

File: preprocessing.py (hunk 1 above)
Comment (amj) [reply preprocessing.py]: Done.

File: main.py (hunk 3 above)
Comment (amj) [reply main.py]: gilding the lily a bit but done :)

File: main.py [hunk 4]
```diff
@@ -118,10 +118,12 @@ def evaluate(
 def selfplay(
         load_file: "The path to the network model files",
         output_dir: "Where to write the games"="data/selfplay",
+        holdout_dir: "Where to write the games"="data/holdout",
```
Comment (Kashomon) [inline main.py]: is this written to local disk?

Comment (amj) [reply main.py]: It's written to wherever you pass it.  This can be a gs:// path

**Pull request merged**